lib/targets/cuda/jitify_helper.cpp

```diff
@@ -0,0 +1,123 @@
+    }
+    
+    if (program_map.find(file) == program_map.end()) {
+
+      std::vector<std::string> options = {
+        "-ftz=true", "-prec-div=false", "-prec-sqrt=false", // match offline optimization options
+        "-remove-unused-globals",                           // remove unused globals to monimize module size
+
+#if __cplusplus >= 201703L
+        "-std=c++17",                                       // use C++17 dialect
+#else
+        "-std=c++14",                                       // use C++14 dialect
+#endif
+
+#ifdef DEVICE_DEBUG
+        "-G",
+#endif
+
+#if CUDA_VERSION >= 11200
+        "-err-no",                                          // display error/warning numbers
+        // disable warnings that are unavoidable
+        "-diag-suppress=64",                                // declaration does not declare anything (anonymous structs in CUB)
+        "-diag-suppress=161"                                // unknown pragmas, e.g., OpenMP
+#endif
+      };
+
+      // add an extra compilation options specific to this instance
+      for (auto option : extra_options) options.push_back(option);
+
+      jitify::Program *program = new jitify::Program(kernel_cache->program(file, 0, options));
+      program_map[file] = program;
+    }
+  }
+
+  qudaError_t launch_jitify(const std::string &file, const std::string &kernel,
+                            const std::vector<std::string> &template_args,
+                            const TuneParam &tp, const qudaStream_t &stream,
+                            std::vector<void*> &arg_ptrs, jitify::detail::vector<std::string> &arg_types,
+                            std::vector<size_t> &arg_sizes, bool use_kernel_arg)
+  {
+    if (arg_ptrs.size() > 1) errorQuda("Unsupported number of kernel arguments = %lu", arg_ptrs.size());
+
+    std::string kernel_file(std::string("kernels/") + file);
+    create_jitify_program_v2(kernel_file);
+
+    std::string kernel_name(std::string("quda::") + kernel);
+    auto instance = program_map[kernel_file]->kernel(kernel_name).instantiate(template_args);
+
+    if (tp.set_max_shared_bytes && device::max_dynamic_shared_memory() > device::max_default_shared_memory()) {
+      instance.set_func_attribute(CU_FUNC_ATTRIBUTE_PREFERRED_SHARED_MEMORY_CARVEOUT, 100);
+      auto shared_size = instance.get_func_attribute(CU_FUNC_ATTRIBUTE_SHARED_SIZE_BYTES);
+      instance.set_func_attribute(CU_FUNC_ATTRIBUTE_MAX_DYNAMIC_SHARED_SIZE_BYTES,
+                                  device::max_dynamic_shared_memory() - shared_size);
+    }
+
+    for (size_t i = 0; i < arg_ptrs.size(); i++) {
+      if (!use_kernel_arg) {
+        auto device_ptr = instance.get_constant_ptr("quda::device::buffer");
+        qudaMemcpyAsync((void*)device_ptr, arg_ptrs[i], arg_sizes[i], qudaMemcpyHostToDevice, stream);
+      }
+    }
+
+    auto configured_instance = instance.configure(tp.grid, tp.block, tp.shared_bytes, device::get_cuda_stream(stream));
```

Comment:
get_stream?

Reply:
1a90dffed

> @maddyscientist @cpviolator Saw this while merging feature/ndeg-twisted-clover with the current GK:
> 
> https://github.com/lattice/quda/blob/a9c7cbdeb7aed146422c0f7333c2ee532c2a1c20/lib/interface_quda.cpp#L3437
> 
> Is this still correct given the separation of `clover_coeff` and `csw` ?

@cpviolator can you take a look?